raiden/tests/integration/rpc/test_assumptions.py

```diff
@@ -59,24 +59,11 @@ def test_call_invalid_selector(deploy_client, blockchain_backend):
     assert result == ''
 
 
-@pytest.mark.parametrize('blockchain_type', ['geth'])
-def test_call_throws(deploy_client, blockchain_backend):
```

Comment:
Why was this test removed? What is the new behavior for a call that throwed?

Reply:
I don't get it, why not?

Reply:
I did not express myself correctly.

I removed this as this is not a good assumption. Any call to a contract function that does not return anything value will also return an empty string, as such it's not a safe thing to check for.

e.g. if `fail()` did not have the require and no throw happened it would still return the empty string.

Reply:
Yes, that is known and it's written here:

https://github.com/raiden-network/raiden/blob/5db35840aebd24744984942884597bc67847f76d/raiden/network/rpc/client.py#L85-L88

That is an assumption about the RPC interface that we rely on and there is no alternative to it, the `eth_call` doesn't have metadata to inform if the call actually failed, and there is no receipt for this RPC method. Currently we rely mostly on the fact that the empty string is returned from an address without code.

Reply:
There is also no way to distinguish a function that returns nothing from the error. That's a dangerous assumption to have. I was wondering if this check is needed.

At the moment it seems the only place that it is used is in `rpc/client.py` in all the locations like this:

https://github.com/raiden-network/raiden/blob/master/raiden/network/rpc/client.py#L1563-1570

So the form the check takes is:

1. trying the call (only for something that does return a value - for other things this check won't work) , 
2. if it's empty string then check if the address has code, if not then we give the no code in address error
3. If the address has code and we get this then only other possible things that happened are:
     i. The contract does not have the function we try to call
     ii. The function had an exception in the outermost call
     iii. The function returns nothing.

Hmm now that I write the above down, I see that even if we change the order of checks we gain nothing and even though I still think it's a very flaky thing to rely on as it relies on the type of function calls and the behaviour of the RPC call it's the best we can now do.

So I will bring back the test, to detect when/if this assumption is broken by a client update.